static/app/components/assistant/guideAnchor.tsx

```diff
@@ -33,216 +33,180 @@ type Props = {
     pathname: string;
     query: Query;
   };
-};
-
-type State = {
-  active: boolean;
-  orgId: string | null;
-  step: number;
-  currentGuide?: Guide;
-};
-
-class BaseGuideAnchor extends Component<Props, State> {
-  state: State = {
-    active: false,
-    step: 0,
-    orgId: null,
-  };
-
-  componentDidMount() {
-    const {target} = this.props;
-    target && registerAnchor(target);
-  }
-
-  componentDidUpdate(_prevProps: Props, prevState: State) {
-    if (this.containerElement.current && !prevState.active && this.state.active) {
-      try {
-        const {top} = this.containerElement.current.getBoundingClientRect();
-        const scrollTop = window.pageYOffset;
-        const centerElement = top + scrollTop - window.innerHeight / 2;
-        window.scrollTo({top: centerElement});
-      } catch (err) {
-        Sentry.captureException(err);
-      }
+}>;
+
+function BaseGuideAnchor({
+  target,
+  onFinish,
+  children,
+  position,
+  offset,
+  containerClassName,
+  to,
+}: Props) {
+  const guideContainerRef = useRef<HTMLSpanElement>(null);
+  const {orgId, forceHide, currentGuide, currentStep} = useLegacyStore(GuideStore);
+
+  const isActive =
+    forceHide !== true && currentGuide?.steps[currentStep]?.target === (target ?? false);
+
+  // Register the anchor in the store
+  useEffect(() => {
+    if (!target) {
+      return () => {};
     }
-  }
 
-  componentWillUnmount() {
-    const {target} = this.props;
-    target && unregisterAnchor(target);
-    this.unsubscribe();
-  }
+    registerAnchor(target);
+    return () => void unregisterAnchor(target);
+  }, [target]);
 
-  unsubscribe = GuideStore.listen(
-    (data: GuideStoreState) => this.onGuideStateChange(data),
-    undefined
-  );
+  // Scroll to the guide on activation
+  useEffect(() => {
+    if (!isActive) {
+      return;
+    }
 
-  containerElement = createRef<HTMLSpanElement>();
+    if (guideContainerRef.current === null) {
+      return;
+    }
 
-  onGuideStateChange(data: GuideStoreState) {
-    const active =
-      (data.currentGuide?.steps[data.currentStep]?.target === this.props.target ??
-        false) &&
-      !data.forceHide;
+    try {
+      const {top} = guideContainerRef.current.getBoundingClientRect();
+      const scrollTop = window.pageYOffset;
+      const centerElement = top + scrollTop - window.innerHeight / 2;
+      window.scrollTo({top: centerElement});
+    } catch (err) {
+      Sentry.captureException(err);
```

Comment:
doesn't Sentry capture this automatically?

Reply:
sure...we should use the try-catch here, but I remember once using `Sentry.capture `exception and getting a feedback that this is automatically captured by Sentry... but I guess that is fine too

Reply:
When would this code crash though? I can only think of a case where getBoundingClientRect would throw, but I dont think this ever happens because you are already early returning if ref.current === null above. It seems like the only real place where this would throw is if we call getBoundingClientRect on a non DOM element, but TSC wouldn't allow you to do that anyways, so it looks like you should be safe just removing this